javascript/core/src/verbs/merge.ts

```diff
@@ -49,7 +69,47 @@ export const mergeStep: ColumnTableStep<MergeArgs> = (
 		}
 	})
 
-	return input.derive({ [to]: func })
+	if (keepOriginalColumns)
+		return unhot
+			? copyTable.derive({ [to]: func })
+			: input.derive({ [to]: func })
+
+	return unhot
+		? copyTable.derive({ [to]: func }).select(not(columns))
+		: input.derive({ [to]: func }).select(not(columns))
+}
+
+function unhotOperation(
```

Comment:
Rather than mutating the inputTable within this function, you should create a new output table and return it, so the function is "pure"